### PR TITLE
Lochen/pipeline component pup

### PR DESCRIPTION
**Pull request by cloga**
# Description
Example update for pipeline component public preview

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: cli/jobs/pipelines-with-components/shakespear_sample_and_word_count_using_spark/README.md [hunk 1]
```diff
@@ -1,17 +0,0 @@
----
-page_type: sample
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/shakespear_sample_and_word_count_using_spark/README.md:2]: confirm: is this a by-design delete?

File: sdk/python/jobs/pipelines/1j_pipeline_with_pipeline_component/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Pipeline component
+## Overview
+When developing complex machine learning  pipeline, there will be sub-pipeline which will use multi-step to doing some task such as data preprocessing, model training. And they can develop and test standalone. We introduce pipeline component, which can group multi-step as component, then you can use them to as single step built complex pipeline, this will help your share your work and better collaborate with team members.
```
Comment (wangchao1230) [inline sdk/python/jobs/pipelines/1j_pipeline_with_pipeline_component/README.md:3]: extra space

File: cli/assets/component/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+$schema: https://azuremlschemas.azureedge.net/latest/pipelineComponent.schema.json
```
Comment (wangchao1230) [inline cli/assets/component/pipeline.yml:1]: MARK: add ci workflows for the newly added samples

File: cli/jobs/pipelines-with-components/shakespear_sample_and_word_count_using_spark/README.md (hunk 1 above)
Comment (cloga) [reply cli/jobs/pipelines-with-components/shakespear_sample_and_word_count_using_spark/README.md]: It is by accident, I already bring back spark component exmaple.

File: sdk/python/jobs/pipelines/1j_pipeline_with_pipeline_component/nyc_taxi_data_regression_with_pipeline_component/nyc_taxi_data_regression_with_pipeline_component.ipynb [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# add meta info when invoke decorator.\n",
+    "@pipeline(display_name='data_pipeline',\n",
+    "    tags={'authoring':'sdk'},\n",
+    "    # version='123',\n",
+    "    description='pipeline component with data prep and transformation defined via yaml-meta.',\n",
+    "    is_deterministic=True\n",
```
Comment (wangchao1230) [inline sdk/python/jobs/pipelines/1j_pipeline_with_pipeline_component/nyc_taxi_data_regression_with_pipeline_component/nyc_taxi_data_regression_with_pipeline_component.ipynb:147]: no is_deterministic for pipeline component?

Comment (cloga) [reply sdk/python/jobs/pipelines/1j_pipeline_with_pipeline_component/nyc_taxi_data_regression_with_pipeline_component/nyc_taxi_data_regression_with_pipeline_component.ipynb:147]: Good question, theoretically, pipeline component shouldn't have this, as it is more like container, this property may not work on it.

**Review by cloga — COMMENTED** [review]
add pipeline component pup sample

**Review by brynn-code — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**